plugin/network_processor.go

```diff
@@ -0,0 +1,258 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+)
+
+type networkInfo struct {
+	IpAddresses []string
+}
+
+// Common interface for IP configuration processing
+type networkProcessor interface {
+	processIPConfigurations(ctx context.Context, ipConfigs []network.InterfaceIPConfiguration, ifName string) (*networkInfo, error)
+	getPublicIPAddress(ctx context.Context, publicIPRef *network.PublicIPAddress, ipConf *network.InterfaceIPConfiguration, ifName string) (string, error)
+	getNetworkInterface(ctx context.Context, interfaceID string) (*network.Interface, string, error)
+}
+
+// Standard VM network processor
+type vmNetworkProcessor struct {
+	clients       *azureClients
+	resourceGroup string
+}
+
+// VMSS network processor
+type vmssNetworkProcessor struct {
+	clients       *azureClients
+	resourceGroup string
+	vmssName      string
+	instanceID    string
+}
+
+func processNetworkProfile(
+	ctx context.Context,
+	interfaces []compute.NetworkInterfaceReference,
+	processor networkProcessor,
+	targetInfo *networkInfo) error {
```

Comment:
Nit, but normally this would look like this:
```suggestion
	targetInfo *networkInfo,
) error {
```